backend/protondrive/protondrive.go

```diff
@@ -0,0 +1,1040 @@
+			}
+		}
+	}
+
+	retReader := io.NopCloser(reader) // the NewLimitedReadCloser will deal with the limit
+
+	// deal with offset
+	if _, err := io.CopyN(io.Discard, retReader, offset); err != nil {
+		return nil, err
+	}
+
+	// deal with limit
+	return readers.NewLimitedReadCloser(retReader, limit), nil
+}
+
+// Update in to the object with the modTime given of the given size
+//
+// When called from outside an Fs by rclone, src.Size() will always be >= 0.
+// But for unknown-sized objects (indicated by src.Size() == -1), Upload should either
+// return an error or update the object properly (rather than e.g. calling panic).
+func (o *Object) Update(ctx context.Context, in io.Reader, src fs.ObjectInfo, options ...fs.OpenOption) error {
+	size := src.Size()
+	if size < 0 {
+		return ErrCanNotUploadFileWithUnknownSize
+	}
+
+	remote := o.Remote()
+	leaf, folderLinkID, err := o.fs.dirCache.FindPath(ctx, o.fs.sanitizePath(remote), true)
+	if err != nil {
+		return err
+	}
+
+	modTime := src.ModTime(ctx)
+	var linkID string
+	var originalSize int64
+	if err = o.fs.pacer.Call(func() (bool, error) {
+		linkID, originalSize, err = o.fs.protonDrive.UploadFileByReader(ctx, folderLinkID, leaf, modTime, in, 0)
+		return shouldRetry(ctx, err)
+	}); err != nil {
+		return err
+	}
+
+	o.id = linkID
+	o.originalSize = &originalSize
+
+	return nil
+}
+
+// Remove an object
+func (o *Object) Remove(ctx context.Context) error {
+	return o.fs.pacer.Call(func() (bool, error) {
+		err := o.fs.protonDrive.MoveFileToTrashByID(ctx, o.id)
+		return shouldRetry(ctx, err)
+	})
+}
+
+// ID returns the ID of the Object if known, or "" if not
+func (o *Object) ID() string {
+	return o.id
+}
+
+// Purge all files in the directory specified
+//
+// Implement this if you have a way of deleting all the files
+// quicker than just running Remove() on the result of List()
+//
+// Return an error if it doesn't exist
+func (f *Fs) Purge(ctx context.Context, dir string) error {
+	root := path.Join(f.root, dir)
+	if root == "" {
+		// we can't remove the root directory, but we can list the directory and delete every folder and file in here
+		return ErrCanNotPurgeRootDirectory
+	}
+
+	folderLinkID, err := f.dirCache.FindDir(ctx, f.sanitizePath(dir), false)
+	if err != nil {
+		return err
+	}
+
+	if err = f.pacer.Call(func() (bool, error) {
+		err = f.protonDrive.MoveFolderToTrashByID(ctx, folderLinkID, false)
+		return shouldRetry(ctx, err)
+	}); err != nil {
+		return err
+	}
+
+	f.dirCache.FlushDir(dir)
+	return nil
+}
+
+// MimeType of an Object if known, "" otherwise
+func (o *Object) MimeType(ctx context.Context) string {
+	return o.mimetype
+}
+
+// Disconnect the current user
+func (f *Fs) Disconnect(ctx context.Context) error {
+	return f.pacer.Call(func() (bool, error) {
+		err := f.protonDrive.Logout(ctx)
+		return shouldRetry(ctx, err)
+	})
+}
+
+// Move src to this remote using server-side move operations.
+//
+// # This is stored with the remote path given
```

Comment:
You need to put a full stop on the end of the sentence to stop `go fmt` insterting these strange `#` symbols!

Reply:
Ahh this solves my long-term question of what's going on with the `go fmt`! Thanks for pointing this out!
Fixed as requested.